pkg/infra/usagestats/usage_stats.go

```diff
@@ -181,9 +181,11 @@ func (uss *UsageStatsService) sendUsageStats(oauthProviders map[string]bool) {
 
 	client := http.Client{Timeout: 5 * time.Second}
 	go func() {
-		if _, err := client.Post(usageStatsURL, "application/json", data); err != nil {
+		resp, err := client.Post(usageStatsURL, "application/json", data)
+		if err != nil {
 			metricsLogger.Error("Failed to send usage stats", "err", err)
 		}
+		defer resp.Body.Close()
```

Comment:
Defer needed here?

Reply:
Good catch, I also noticed you shouldn't try to close here if an error was returned.